src/blk/BlockDevice.h

```diff
@@ -286,6 +287,7 @@ class BlockDevice {
   virtual int flush() = 0;
   virtual bool try_discard(interval_set<uint64_t> &to_release, bool async=true) { return false; }
   virtual void discard_drain() { return; }
+  virtual int discard(uint64_t offset, uint64_t len) { return 0; }
```

Comment:
1. `return -ENOSUPP` please.
2. Why is new call needed? `try_discard(to_release, false)` works the same.

Reply:
> Why is new call needed? try_discard(to_release, false) works the same.

ok, I am trying to reuse `try_discard(to_release, false)`

Reply:
done